pkg/ipmi/ipmi.go

```diff
@@ -173,3 +178,62 @@ func (i *IPMI) ShutoffWatchdog() error {
 
 	return nil
 }
+
+func (i *IPMI) getDeviceID() ([]byte, error) {
+	req := &req{}
+	req.msg.netfn = _IPMI_NETFN_APP
+	req.msg.cmd = _BMC_GET_DEVICE_ID
+
+	recv, err := i.sendrecv(req)
+	if err != nil {
+		return nil, err
+	}
+
+	return recv, nil
+}
+
+func (i *IPMI) setGlobalEnables(enables byte) error {
+	req := &req{}
+	req.msg.netfn = _IPMI_NETFN_APP
+	req.msg.cmd = _BMC_SET_GLOBAL_ENABLES
+	req.msg.data = unsafe.Pointer(&enables)
+	req.msg.dataLen = 1
+
+	_, err := i.sendrecv(req)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (i *IPMI) getGlobalEnables() ([]byte, error) {
+	req := &req{}
+	req.msg.netfn = _IPMI_NETFN_APP
+	req.msg.cmd = _BMC_GET_GLOBAL_ENABLES
+
+	recv, err := i.sendrecv(req)
+	if err != nil {
+		return nil, err
+	}
+
+	return recv, nil
+}
+
+func (i *IPMI) EnableSEL() error {
+	// Check if SEL device is supported or not
+	if data, err := i.getDeviceID(); err != nil || (data[6]&0x04) == 0 {
```

Comment:
Is this needed? I don't see the `ipmitool` doing this. Is it documented somewhere?

Reply:
It base on IPMI spec v2, BMC global enables command,
to check and set  BMC SEL enabled.

Reply:
Where does it mention that getting the device id is required to enable the System Event Logging?

Reply:
To check BMC information of additional logical device functionality 
which the SEL device support within.

Reply:
Isn't the getGlobalEnables() enough?

I read:

```
Enable/Disable Mask
These sixteen bytes form a 128-bit bitfield where each bit returns the
enable/disable of a particular command value under the given NetFn. If
a command is not supported at all, a 0b will be returned.
```

I need to dig a bit more.

Reply:
Ok, looks good.